Add troubleshooting, data updates and known limitations to the bosch_alarm integration (#38376)
home-assistant/home-assistant.github.io: source/_integrations/bosch_alarm.markdown

ha_iot_class: Local Push
ha_config_flow: true
ha_codeowners:
  - "@mag1024"

>   - "@sanjay900"

Reviewer: ```suggestion
  - '@mag1024'
  - '@sanjay900'
```

The integration will prompt for the required passcodes, which depend on the panel type.

| Panel    | Code       |
| -------- | ---------- |
| Solution | User [^2]  |
| B Series | Automation |
| G Series | Automation |
| AMAX     | Both       |

[^2]: The user needs to have the "master code functions" authority if you wish to interact with history events.

{% important %}
Since the _Mode 2_ automation user has "superuser" privileges, it bypasses the regularly configured alarm pin: you will _not_ be prompted for a _User_ code when arming/disarming through the integration.
{% endimportant %}

## Data updates

The Bosch Alarm Panel {% term integration %} fetches data from the device every 30 seconds.
Newer devices and firmware revisions have the possibility to push data instead of needing to rely on {% term polling %}.
At startup, the integration checks whether your panel supports push data updates and falls back to {% term polling %} if not.

## Troubleshooting

### Issues with Bosch Solution 2000/3000/4000 panels

> 

Reviewer: How do I know that a misconfiguration is causing the issue? What are the symptoms?

We have found that some panels end up with a configuration on them that is incompatible with the integration. The easiest way to solve this is to follow the full reset and restore procedure outline below.

> #### Full Reset & Restore Procedure

Reviewer: ```suggestion
#### Resetting the configuration and restoring the integration
```

> 1. Update firmware (Recommended)

Reviewer: Please add a period to the end of each sentence.

2. Back up the existing configuration
   - Connect to the panel via A-Link Plus
   - Perform an Upload of the panel configuration
   - Save the configuration to your computer
3. Default the control panel
   - Press the default/reset button on the panel
   - Use installer code 1234
   - Set:
     - Location 0081 = 3 (Enables IP module mode)
     - Location 4456 = 4 (Enables RSC+ communication)
   - Set date and time using master code 25806#
4. Initial Home Assistant test
   - Wait 2 to 5 minutes after resetting the panel
   - Connect Home Assistant to the panel using its IP address
   - Home Assistant should connect using default config and show panel status
5. Restore your original configuration
   - Reconnect to the panel using A-Link Plus
   - Modify zones, outputs, and user codes to match original setup
   - Save and Download the updated config to the panel
   - Wait 2 to 5 minutes
6. Reconnect to Home Assistant
   - Open Home Assistant
   - The integration should now detect the updated configuration
   - All relevant entities (zones, partitions, outputs) should appear automatically

#### Restarting a panels network stack

> We have found that the Solution panels have a bug where they can get into a state where the network module stops letting us use the _Mode 2_ API.

Reviewer: How do I know this is the issue? What are the symptoms? Is there an error message?

   - Download firmware from the Bosch Security website
2. Set IP Address
   - Configure the panel’s IP settings using RPS or via the codepad
   - Use DHCP for initial setup (optional), or assign a static IP
3. Configure the automation device
   - In RPS:
     - Set Automation Device to Mode 2 (Bosch Standard Protocol)
     - Set your automation passcode (used for Home Assistant authentication)
4. Wait for changes to apply
   - Wait 2 to 5 minutes for the panel to reboot and apply settings
5. Connect to Home Assistant
   - Use panel’s IP address or allow auto-discovery
   - Enter the automation passcode in the Home Assistant configuration
   - Home Assistant should connect and display panel status, zones, and partitions

#### TLS issues

Some older firmwares for these panels use outdated certificates that are no longer trusted by Home Assistant. If you have issues connecting and see a TLS error in your logs, update the firmware on your panel.

## Known limitations

The integration does not provide the ability to configure the panel, which can instead be done via the configuration utility for your panel.

> Some older firmwares for the Solution / AMAX series panels only support a single connection at a time. If you try to have a cloud connection and use the integration on these panels simultaneously, the panel's network stack can lock up, and the integration will stop working.

Reviewer: ```suggestion
- The integration does not allow you to configure the panel; you can instead do this via the panel's configuration utility.
- Some older firmware versions for the Solution / AMAX series panels only support a single connection at a time. If you try to have a cloud connection and use the integration on these panels simultaneously, the panel's network stack can lock up, and the integration will stop working.
```
If these are two entires independent points, I would use a list.

The primary means of authentication for the _Mode 2_ API is the _Automation_ passcode. It needs to be at least 10 characters long, and it is different from the _User_ code -- a shorter numeric pin used to arm/disarm the panel.
The integration will prompt for the required passcodes, which depend on the panel type.

| Panel    | Code       |
| -------- | ---------- |
| Solution | User [^2]  |
| B Series | Automation |
| G Series | Automation |
| AMAX     | Both       |

[^2]: The user needs to have the "master code functions" authority if you wish to interact with history events.

{% important %}
Since the _Mode 2_ automation user has "superuser" privileges, it bypasses the regularly configured alarm pin: you will _not_ be prompted for a _User_ code when arming/disarming through the integration.
{% endimportant %}

## Data updates

> The Bosch Alarm Panel {% term integration %} fetches data from the device every 30 seconds.

Reviewer: ```suggestion
The **Bosch Alarm** {% term integration %} fetches data from the device every 30 seconds.
```
I know it is spelled differently further up, but isn't the integration called Bosch Alarm?

Newer devices and firmware revisions have the possibility to push data instead of needing to rely on {% term polling %}.
At startup, the integration checks whether your panel supports push data updates and falls back to {% term polling %} if not.

## Troubleshooting

### Issues with Bosch Solution 2000/3000/4000 panels

We have found that some panels end up with a configuration on them that is incompatible with the integration. 
When this happens, you will get a connection error when attempting to communicate with the panel, even if the credentials are correct.
If you continue to have connection errors even after rebooting your panel, then you have an issue with your configuration.
The easiest way to solve this is to follow the full reset and restore procedure outline below.

#### Resetting the configuration and restoring the integration

1. Update firmware (Recommended).
   1. Download and install the latest firmware for the control panel and IP module from the Bosch Security website.
2. Back up the existing configuration.

>    1. Connect to the panel via A-Link Plus.

Reviewer: A-Link Plus hasn't been mentioned in the documentation before. Do people know what this is?

Author: Yeah so for the solution and amax panels, its the software used for programming and configuring the panels
For the B/G series, you have RPS

Reviewer: Is it worth adding a link to their documentation?

Author: I would assume someone who has configured their panel before would know what it is, but if someone installed the panel for them they may not know it exists - but then in that scenario they probably shouldn't be reconfiguring it anyways

>    2. Perform an Upload of the panel configuration.

Reviewer: ```suggestion
   2. Perform an upload of the panel configuration.
```
Is Upload capital because it is a UI string? In this case, it should be bold. We use bold to mark UI strings.

Author: ah yes it is a ui string

Reviewer: ok, but if it's phrased like this is does not need to be bold. only when we explicitly refer to the UI string, then it's capital and bold (if the actual string starts with a capital). If it is phrased like this, as a normal verb, then it should not start with a capital.

Author: ah right, ill make it capital + bold since it is refering to a button in the configuration tool

Reviewer: haha :smile: I'm sorry if my comment led to confusion. I think it should just be non-caps like suggested in this case. bold and capital is when referring explicitly to the string. For example: To do XYZ, in the top-right corner, select **Upload**. Or like you did in line 133.

Author: right that makes sense

   3. Save the configuration to your computer.
3. Default the control panel.
   1. Press the default/reset button on the panel.
   2. Use installer code 1234.
   3. Set:
     1. Location 0081 = 3 (Enables IP module mode)

>      2. Location 4456 = 4 (Enables RSC+ communication)

Reviewer: ```suggestion
      1. Location 0081 = 3 (Enables IP module mode)
      2. Location 4456 = 4 (Enables RSC+ communication)
```

   4. Set date and time using master code `25806#`.
4. Initial Home Assistant test.
   1. Wait 2 to 5 minutes after resetting the panel.
   2. Connect Home Assistant to the panel using its IP address.

>    3. Home Assistant should connect using default config and show panel status.

Reviewer: ```suggestion
   3. Home Assistant should connect using default configuration and show panel status.
```

5. Restore your original configuration.
   1. Reconnect to the panel using A-Link Plus.

>    2. Modify zones, outputs, and user codes to match original setup.

Reviewer: ```suggestion
   2. Modify zones, outputs, and user codes to match the original setup.
```

>    3. Save and Download the updated config to the panel.

Reviewer: Are these starting with a capital letter because they are UI strings? In this case, they should be bold. We use bold to mark UI strings.

   4. Wait 2 to 5 minutes.
6. Reconnect to Home Assistant.
   1. Open Home Assistant.
   2. The integration should now detect the updated configuration.
   3. All relevant entities (zones, partitions, outputs) should appear automatically.

#### Restarting a panels network stack

We have found that the Solution panels have a bug where they can get into a state where the network module stops letting us use the _Mode 2_ API.

> If your configuration information is correct but you are still unable to connect to your panel, than you are likley encountering this bug.
> This can be resolved by restating the network modules, which can be done with the following steps.

Reviewer: ```suggestion
If your configuration information is correct, but you are still unable to connect to your panel, then you are likely encountering this bug.
This can be resolved by restarting the network modules, which can be done with the following steps.
```

##### Resetting network module 1

Using the codepad, enter your master code, followed by the `[9][4][1]` and `[#]` keys.

##### Resetting network module 2

Using the codepad, enter your master code, followed by the `[9][4][2]` and `[#]` keys.

### Issues with the Bosch B/G Series (B3512/B4512/B5512/B8512/B9512)

The following procedure can be used to configure the panel correctly so that it will work with the integration.

#### Panel configuration procedure

> 

Reviewer: same in this section. Please end all sentences with a period.
